pallets/acurast/common/src/types/bounded_attestation.rs

```diff
@@ -662,3 +683,47 @@ impl From<asn::VerifiedBootState> for VerifiedBootState {
         }
     }
 }
+
+#[derive(RuntimeDebug, Encode, Decode, MaxEncodedLen, TypeInfo, Clone, PartialEq, Eq)]
+pub struct BoundedAttestationApplicationId {
+    pub package_infos: PackageInfoSet,
+    pub signature_digests: SignatureDigestSet,
+}
+
+impl<'a> TryFrom<asn::AttestationApplicationId<'a>> for BoundedAttestationApplicationId {
+    type Error = ();
+
+    fn try_from(value: asn::AttestationApplicationId<'a>) -> Result<Self, Self::Error> {
+        Ok(Self {
+            package_infos: value
+                .package_infos
+                .map(|package_info| BoundedAttestationPackageInfo::try_from(package_info))
+                .collect::<Result<Vec<BoundedAttestationPackageInfo>, ()>>()?
+                .try_into()
+                .map_err(|_| ())?,
+            signature_digests: value
+                .signature_digests
+                .map(|digest| Digest::try_from(digest.to_vec()))
+                .collect::<Result<Vec<Digest>, ()>>()?
+                .try_into()
+                .map_err(|_| ())?,
+        })
+    }
+}
+
+#[derive(RuntimeDebug, Encode, Decode, MaxEncodedLen, TypeInfo, Clone, PartialEq, Eq)]
+pub struct BoundedAttestationPackageInfo {
+    pub package_name: PackageName,
+    pub version: i64,
+}
+
+impl<'a> TryFrom<asn::AttestationPackageInfo<'a>> for BoundedAttestationPackageInfo {
+    type Error = ();
```

Comment:
Having a proper error here and in other places would be nice. But nothing blocking.